src/ui/public/vis/editors/default/vis_type_agg_filter.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { AggType } from '../../../agg_types';
+import { aggTypeFilters } from '../../../agg_types/filter';
+import { AggConfig, Vis } from '../../../vis';
+
+import { propFilter } from '../../../filters/_prop_filter';
+
+const filterByName = propFilter('name');
+
+/**
+ * This filter checks the defined aggFilter in the schemas of that visualization
+ * and limits available aggregations based on that.
+ */
+aggTypeFilters.register((aggType: AggType, vis: Vis, aggConfig: AggConfig) => {
+  return filterByName([aggType], aggConfig.schema.aggFilter).length === 1;
```

Comment:
I had to scratch my head for a few minutes to understand why we were checking for strict equality with 1. And then I realized that it was simply because there is a single element in `[aggType]`. 😄 I feel like this is a bit misleading (and unnecessary). It might be clearer to do this instead:

```js
aggTypeFilters.register((aggType: AggType, vis: Vis, aggConfig: AggConfig) => {
  const doesFilterAllowAggType = filterByName([aggType], aggConfig.schema.aggFilter).length !== 0;
  return doesFilterAllowAggType;
});
```